server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -432,16 +438,64 @@ public IngestInfo info() {
         return processorMetrics;
     }
 
+    // visible for testing
     static ClusterState innerPut(PutPipelineRequest request, ClusterState currentState) {
         IngestMetadata currentIngestMetadata = currentState.metadata().custom(IngestMetadata.TYPE);
+
+        BytesReference pipelineSource = request.getSource();
+        if (request.isVersionedUpdate()) {
+            var currentPipeline = currentIngestMetadata != null ? currentIngestMetadata.getPipelines().get(request.getId()) : null;
+            if (currentPipeline == null) {
+                throw new IllegalStateException(String.format(
+                    Locale.ROOT,
+                    "version conflict, required version [%s] for pipeline [%s] but no pipeline was found",
+                    request.getVersion(),
+                    request.getId()
```

Comment:
Currently, any of these exceptions for OCC violations that are thrown from within the cluster state update task submitted above [on line 376](https://github.com/elastic/elasticsearch/pull/78551/files#diff-055eff0044e6cbbab2fdd7c5a0fb5bf06afb55141c00c6d89c598ef6870e908aR376) are propagated to the client as HTTP 500 errors. I was unclear on the error handling chain there, but I believe these should be returned to the client as HTTP 400 errors instead.